qiskit_experiments/calibration_management/basis_gate_library.py

```diff
@@ -289,3 +289,120 @@ def default_values(self) -> List[DefaultCalValue]:
                     defaults.append(DefaultCalValue(value, param.name, tuple(), name))
 
         return defaults
+
+
+class FixedFrequencyTransmonCR(FixedFrequencyTransmon):
+    """A fixed-frequency transmon gate library with echoed cross-resonance gates.
+
+    This library extends the FixedFrequencyTransmon library by adding support for
+    cross-resonance gates.
+    """
+
+    __default_values__ = {
+        "duration": 160,
+        "amp": 0.5,
+        "β": 0.0,
+        "cr_duration": 640,
+        "cr_width": 384,
+        "cr_σ": 64,
+        "cr_amp": 0.5,
+        "rot_amp": 0.0,
+    }
+
+    def __init__(
+        self,
+        echo: bool = True,
+        rotary: bool = True,
+        basis_gates: Optional[List[str]] = None,
+        default_values: Optional[Dict] = None,
+        link_parameters: bool = True,
+    ):
+        """Setup the schedules.
+
+        Args:
+            echo: A boolean to indicate if an echo is used in the CR gate schedule.
+                The default is True.
+            rotary: A boolean to indicate if a rotary tones are used in the CR gate.
+            basis_gates: The basis gates to generate.
+            default_values: Default values for the parameters this dictionary can contain
+                the following keys: "duration", "amp", "β", and "σ". If "σ" is not provided
+                this library will take one fourth of the pulse duration as default value.
+            link_parameters: if set to True then the amplitude and DRAG parameters of the
+                X and Y gates will be linked as well as those of the SX and SY gates.
+        """
+        self._echo = echo
+        self._rotary = rotary
+        super().__init__(basis_gates, default_values, link_parameters)
+
+    @property
+    def __supported_gates__(self) -> Dict[str, int]:
+        """The gates that this library supports."""
+        gates = super().__supported_gates__
+        gates.update({"cr": 2, "cr90p": 2, "cr90m": 2})
+        return gates
+
+    def _build_schedules(self, basis_gates: Set[str]) -> Dict[str, ScheduleBlock]:
+        """Build the schedules of the library."""
+        if "x" not in basis_gates:
+            raise CalibrationError("x gate is required to build cross-resonance schedules.")
+
+        schedules = super()._build_schedules(basis_gates)
+
+        target = pulse.DriveChannel(Parameter("ch1"))
+        cr_chan = pulse.ControlChannel(Parameter("ch0.1"))
+
+        cr_amp = Parameter("cr_amp")
```

Comment:
We need to calibrate phase too. This means, you need
```python
cr_amp = Parameter("cr_amp") 
cr_phase = Parameter("cr_phase")

complex_amp = cr_amp * np.exp(1j*cr_phase)  # this is parameter expression
```
I really want to show the parameter in the phasor format in the calibration database since the context of what we are actually calibrating is missing with complex format.

Reply:
Yes, I find amp and phase easier to reason about separately as well. `sx` in `FixedFrequencyTransmon` does not have a separate phase parameter either. Some of this might be addressable by reporting magnitude and phase of complex numbers separately in tables showing parameter values.